File: LeetCode/Medium/1233-Remove_Sub-Folders_From_The_Filesystem/1233-RemoveSub-FoldersFromTheFilesystem.ex
defmodule Solution do
  @spec remove_subfolders(folder :: [String.t]) :: [String.t]
  def remove_subfolders(folder) do
    sorted_folder = Enum.sort(folder)

    final_ans = Enum.reduce(Enum.drop(sorted_folder, 1), [Enum.at(sorted_folder, 0)], fn path, acc ->
      prev_folder = "#{hd(acc)}/"

      if not String.starts_with?(path, prev_folder) do
        [path | acc]
      else
        acc
      end
    end)

    final_ans
  end
end

File: LeetCode/Medium/1233-Remove_Sub-Folders_From_The_Filesystem/1233-RemoveSub-FoldersFromTheFilesystem.js
/**
 * @param {string[]} folder
 * @return {string[]}
 */
var removeSubfolders = function (folder) {
    folder.sort();

    let ans = [folder[0]];

    for (let i = 1; i < folder.length; i++) {
        if (!folder[i].startsWith(ans[ans.length - 1] + '/'))
            ans.push(folder[i])
    }

    return ans;
};